overlord/devicestate/devicestate.go

```diff
@@ -223,3 +225,61 @@ func ProxyStore(st *state.State) (*asserts.Store, error) {
 
 	return a.(*asserts.Store), nil
 }
+
+// refreshScheduleManaged returns true if the refresh schedule of the
+// device is managed by an external snap
+func refreshScheduleManaged(st *state.State) bool {
```

Comment:
no strong opinion either way, but this could be moved to snapstate in the new autorefresh.go file and keep only CanSetRefreshScheduleManaged as the hook?

Reply:
I like your suggestion, I think:
```
        snapstate.CanAutoRefresh = canAutoRefresh
-       snapstate.RefreshScheduleManaged = refreshScheduleManaged
+       snapstate.CanSetRefreshScheduleManaged = CanSetRefreshScheduleManaged
```
alone is worth it, nice and symmetrical.